src/containers/Tasks/SubmitTerms/SubmitTerms.tsx

```diff
@@ -117,29 +121,61 @@ class SubmitLegitimations extends React.Component<Props, State> {
     })
   }
 
+  private updateQuote(quote: sdk.IQuote): void {
+    const { quoteData } = this.state
+    if (quoteData !== quote) {
+      this.setState({ quoteData: quote })
+    }
+  }
+
   private sendClaim(): void {
     const {
       getAttestedClaims,
       enablePreFilledClaim,
       receiverAddresses,
       onFinished,
     } = this.props
-    const { claim, selectedDelegation, withPreFilledClaim } = this.state
+    const {
+      claim,
+      selectedDelegation,
+      withPreFilledClaim,
+      quoteData,
+    } = this.state
 
     if (enablePreFilledClaim && !withPreFilledClaim) {
       delete claim.contents
     }
+    const selectedIdentity: sdk.Identity = Wallet.getSelectedIdentity(
+      PersistentStore.store.getState()
+    ).identity
 
-    AttestationWorkflow.submitLegitimations(
-      claim,
-      getAttestedClaims(),
-      receiverAddresses,
-      selectedDelegation
-    ).then(() => {
-      if (onFinished) {
-        onFinished()
-      }
-    })
+    if (!selectedIdentity) {
+      throw new Error('No identity selected')
+    }
+    if (quoteData) {
+      AttestationWorkflow.submitTerms(
+        claim,
+        getAttestedClaims(),
+        receiverAddresses,
+        selectedDelegation,
+        sdk.Quote.createAttesterSignature(quoteData, selectedIdentity)
+      ).then(() => {
+        if (onFinished) {
+          onFinished()
+        }
+      })
+    } else {
+      AttestationWorkflow.submitTerms(
+        claim,
+        getAttestedClaims(),
+        receiverAddresses,
+        selectedDelegation
+      ).then(() => {
+        if (onFinished) {
+          onFinished()
+        }
+      })
+    }
```

Comment:
again, you could reduce duplication like this:
```suggestion
    const quote = quoteData ? sdk.Quote.createAttesterSignature(quoteData, selectedIdentity) : undefined
      AttestationWorkflow.submitTerms(
        claim,
        getAttestedClaims(),
        receiverAddresses,
        selectedDelegation,
        quote
      ).then(() => {
        if (onFinished) {
          onFinished()
        }
      })
```